core/src/test/scala/org/apache/spark/serializer/KryoSerializerSuite.scala

```diff
@@ -328,6 +334,52 @@ class KryoSerializerSuite extends SparkFunSuite with SharedSparkContext {
     assert(!ser2.getAutoReset)
   }
 
+  test("output reuse with JavaSerializer") {
```

Comment:
Since this is a regression test for a pretty subtle bug, maybe we should add comments to provide more context for future readers.  Could you update this to link to the issue that I opened against Kryo (https://github.com/EsotericSoftware/kryo/issues/312)?